lib/SqlString.js

```diff
@@ -24,6 +28,14 @@ SqlString.escapeId = function escapeId(val, forbidQualified) {
     }
 
     return sql;
+  } else if (val && typeof val.toSqlString === 'function') {
+    // If it corresponds to an identifier token, let it through.
+    var sqlString = val.toSqlString();
+    if (QUALIFIED_ID_REGEXP.test(sqlString)) {
+      return sqlString;
+    } else {
+      throw new TypeError();
```

Comment:
Can this provide some kind of message / description that will assist the dev who encounters this thrown error?